### PR TITLE
feat: add support for missing value interpretation

### LINKED ISSUE #426: Default expression does not work
Hi, 

We create a table with multiple fields, and add bgUid field with default value: GENERATE_UUID(), when jsonWriter.appendRows, field bgUid is not included in json list.  We assume one UUID should be generated as the default value, while the value we got is null.  

After filter the field from TableMetadata, default values will take effect. This is not expected. It is expected that when one field is not defined, the default value will take effect. And that is also what the bigquery API do.

Code to filter field:
```
  const table = bigquery.dataset(datasetId).table(tableId);
  const [metadata] = (await table.getMetadata()) as [TableMetadata, unknown];
  const { schema } = metadata as { schema: Schema };
  schema.fields.filter(field=> field.name != 'bgUid');
```

Thanks,

### PR DESCRIPTION
**Pull request by alvarowolfx**
Fixes #426  🦕

### CODE REVIEW

File: system-test/managed_writer_client_test.ts [hunk 1]
```diff
@@ -554,6 +554,105 @@ describe('managedwriter.WriterClient', () => {
...
+      .dataset(datasetId)
+      .createTable(tableId + '_default_values', {schema: updatedSchema});
+    const parent = `projects/${projectId}/datasets/${datasetId}/tables/${table.id}`;
+
+    const storageSchema =
+      adapt.convertBigQuerySchemaToStorageTableSchema(updatedSchema);
+    const protoDescriptor: DescriptorProto =
+      adapt.convertStorageSchemaToProto2Descriptor(storageSchema, 'root');
+
+    // Row 1
```
Comment (shollyman) [inline system-test/managed_writer_client_test.ts:592]: These sample rows are confusing, as the row data isn't compatible with the `updatedSchema` above.  If you're wanting empty rows that's fine, just describe what you're doing so the interaction is clearer.

**Review by shollyman — COMMENTED** [review]
Does this handwritten allow MVI changes when writes have already begun, or are the options locked at the outset only?

**Comment by alvarowolfx** [comment]
> Does this handwritten allow MVI changes when writes have already begun, or are the options locked at the outset only?

Both `Writer` and `JSONWriter` have the methods `setDefaultMissingValueInterpretation` and `setMissingValueInterpretations` to change MVI after the writer is already started.

File: system-test/managed_writer_client_test.ts (hunk 1 above)
Comment (alvarowolfx) [reply system-test/managed_writer_client_test.ts]: I've pushed some updates to the test to make the full schema more explicit. The idea here is to send rows with some empty values and show that they are being filled or not accordingly to the MVI settings.

File: src/managedwriter/writer.ts [hunk 2]
```diff
@@ -72,6 +126,28 @@ export class Writer {
     }
   }
 
+  /**
+   * Update how missing values are interpreted by for the given stream.
```
Comment (feywind) [inline src/managedwriter/writer.ts:130]: Another 'by'

**Review by feywind — APPROVED** [review]

File: src/managedwriter/writer.ts [hunk 3]
```diff
@@ -24,9 +24,60 @@ type ProtoData =
...
+export interface WriterOptions {
+  /** The stream connection to the BigQuery streaming insert operation. */
+  connection: StreamConnection;
+
+  /** The proto descriptor for the stream. */
+  protoDescriptor: IDescriptorProto;
+
+  /**
+   * Controls how missing values are interpreted by for a given stream.
+   * `missingValueInterpretations` set for individual colums can override the default chosen
```
Comment (shollyman) [inline src/managedwriter/writer.ts:44]: nit: s/colums/columns/

**Review by shollyman — APPROVED** [review]
Thanks for the updates, comment nits seem to be the last bit to address.

I do wish the assertion support was better here and isnull/notnull were first class assertions, but alas.

**Pull request merged**